go/arrow/flight/flightsql/client.go

```diff
@@ -450,6 +450,31 @@ func (c *Client) PrepareSubstrait(ctx context.Context, plan SubstraitPlan, opts
 	return parsePreparedStatementResponse(c, c.Alloc, stream)
 }
 
+func (c *Client) LoadPreparedStatementFromResult(result *pb.ActionCreatePreparedStatementResult) (*PreparedStatement, error) {
```

Comment:
rather than exposing the `pb.ActionCreatePreparedStatementResult` directly, please add a type for it in `types.go` with the other protobuf aliasing we do instead. We try to avoid using the protobuf types *directly* in the interface